### ISSUE TITLE
Color Selection doesn't match the design in Figma

### ISSUE THREAD

**Issue by FlicHollis** [google/site-kit-wp#5751]
![Colour selection](https://user-images.githubusercontent.com/93320881/186766218-d65d8341-9c3a-4708-b952-7223f9eccfbb.png)

---------------
_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* For the color selection in Thank with Google settings:
    * The color circle component should be have the same border width as the 2 buttons above it on the Thank with Google setting screen
    * The circles should be larger, in-line with the sizes in the [Figma design](https://www.figma.com/file/t5zDpQ28Lo7ZyRcOpGdqgu/TwG?node-id=2343%3A12526)
    * Circles shouldn't have a circle around them when they're unselected/unfocuses
    * In the original design the selected state should include an outline in the primary selected color, and not green for all.
        * This step is optional, but if possible preferred.

## Implementation Brief

- In `assets/js/components/ImageRadio.js`:
    - Extend the `<ImageRadio />` component so that any additional props passed to the component can be accessed via a spreaded `otherProps` prop, i.e. add a `…otherProps` to ImageRadio [here](https://github.com/google/site-kit-wp/blob/a604447afae4cc1b4c3f0760889dd15985af026d/assets/js/components/ImageRadio.js#L48).
	- Pass `otherProps` to the `div.mdc-image-radio__content` element as spreaded so that any passed props with `otherProps` is passed to the `div.mdc-image-radio__content` element.
- In `assets/js/modules/thank-with-google/util/settings.js`:
    - Update the `getColorThemes` function so that each of the object in the returned array includes a new property named `colorCode` with the color hexcode as the value (the hexcode can be obtained from the Figma design).
- In `assets/js/modules/thank-with-google/components/common/ColorRadio.js`:
    - Locate the `<ImageRadio />` component.
	- Pass a new prop `style` with an object as the value. The object should have a property named `outlineColor` with the `colorCode` (returned from `getColorThemes`) as the value, e.g.
	```js
	<ImageRadio
		{ ...OTHER PROPS }
		style={ { outlineColor: colorCode } }
	/>
	```
- In `assets/sass/components/thank-with-google/_googlesitekit-twg-color-radio.scss`:
    - For the `.googlesitekit-plugin .googlesitekit-twg-color-radio__options .googlesitekit-image-radio .image-radio .mdc-image-radio__background .mdc-image-radio__content` selector:
	    - Set `height` to `44px`.
	    - Set `outline` to `none`.
		- Set `width` to `44px`.
	- For the `.googlesitekit-plugin .googlesitekit-twg-color-radio__options .googlesitekit-image-radio .image-radio .mdc-image-radio__background .mdc-image-radio__content[aria-checked=true]` selector, set `outline` to `0.1875rem solid`.

### Test Coverage

- No tests need to be added/updated.
- Update VRT references if required.

## QA Brief

* Ensure that the `twgModule` feature flag is enabled in the tester plugin.
* Go to the Thank with Google customisation screen.
* Scroll down to the `Color` section.
* Verify that the color selections match the Figma designs.
* Verify that selecting a color displays an outline around it with the same color.
* Verify that the outline width is same as Figma.
* Verify that the gap between the color circle and the outline is same as Figma.

## Changelog entry

* Fix color selection outline in Thank with Google settings.

**Comment by tofumatt** [comment]
Looks good, but the `contentProps` approach isn't my favourite, just because it seems a bit less intuitive/ergonomic. Any reason not to accept either:

1. a more limited number of props to pass through (eg. just `style`) rather than `contentProps`, or
2. add a `…otherProps` to `ImageRadio` [here](https://github.com/google/site-kit-wp/blob/a604447afae4cc1b4c3f0760889dd15985af026d/assets/js/components/ImageRadio.js#L48) and pass those through to the [div in question here](https://github.com/google/site-kit-wp/blob/a604447afae4cc1b4c3f0760889dd15985af026d/assets/js/components/ImageRadio.js#L83-L91)

That way the API'd be a bit more straightforward:

```js
 <ImageRadio
 	{ ...OTHER PROPS }
	style={ { outlineColor: colorCode } }
 />
```

**Comment by nfmohit** [comment]
@tofumatt Very fair point, thank you for raising it!

The only reason I used `contentProps` because we are passing the props to the [child div](https://github.com/google/site-kit-wp/blob/a604447afae4cc1b4c3f0760889dd15985af026d/assets/js/components/ImageRadio.js#L83-L91) and not [the wrapping/parent div](https://github.com/google/site-kit-wp/blob/a41ec5332b3949e6b21731715b63e9dff08aaa16/assets/js/components/ImageRadio.js#L65-L68).

I was worried if using a generic prop like `style` or `otherProps` make it obvious that the props would be applied to the wrapping/parent div. What do you think?

**Comment by tofumatt** [comment]
I looked through the component and thought of that, but I think it's an okay API.

I think for more "basic"/generic/UI components like a `Chip`, `Radio`, etc., a developer should expect that the props they pass will affect the "top-level" component.

By default, in general, that's a good pattern to use.

But in the case case of something very specific and not at all generic, like the `ImageRadio` component—it's such a specific component that doesn't have an obvious generic HTML/UI analogue that having its props not map 1:1 with the output HTML is fine.

Basically, for more specialised components like this, I think it's a fine trade-off if it otherwise makes them more usable and fulfils the common use-case. I think here it does, so it's fine 🙂 

------

On that note, IB looks good 👍🏻 

IB ✅

**Comment by nfmohit** [comment]
Understood, thank you, @tofumatt!

**Comment by mohitwp** [comment]
### QA Update ❌ 

As per AC color selections should match Figma design.
@nfmohit I found that in Figma Design 'Color selection' icons have proper circle design. In actual implementation, 'Color selection' icons are not in proper circle shape. Above area of it have hexagonal shape.

<details>

Figma

![image](https://user-images.githubusercontent.com/94359491/190640016-5781c0a5-588a-4ab5-b1ae-ade593575fd5.png)

Implementation-

![image](https://user-images.githubusercontent.com/94359491/190640097-bfbcbad0-48c2-45d3-a339-0dab483d98aa.png)

</details>

**Comment by nfmohit** [comment]
@mohitwp That's odd. May I know which browser are you seeing this in because I can't seem to replicate it on my end (Chrome 105.0.5195.125 on macOS Monterey 12.6)? Here's a screenshot:
![image](https://user-images.githubusercontent.com/20284937/190697339-7c443a03-e76c-49c3-b37e-367f5aa66823.png)

Also, I'd like to point out that the PR attached to this issue doesn't have any effect on the shape of the color selection icons. It only changes the color of the pre-existing circles.

**Comment by mohitwp** [comment]
@nfmohit Looks like issue producing in only my system. I tried another system having same view port and configuration - Windows 11, Browser- chrome , firefox and edge. I'm not able to reproduce this issue on another system and BrowserStack. 

### QA Update :

Verified- 

- Verified that color selection matches the Figma design.
- Verified that selecting a color displays an outline around it with the same color.
- Verified on Chrome. Safari, Edge and Firefox.

<details>

![image](https://user-images.githubusercontent.com/94359491/191005555-03b78b9b-b220-4961-94ea-48d3cb63b61d.png)

![image](https://user-images.githubusercontent.com/94359491/191005590-2b64ddf7-4fd4-47fd-a3d9-2fafef68836c.png)

</details>

**Comment by aaemnnosttv** [comment]
## Approval ⚠️ 

There is a minor inconsistency in the implementation and the design, specifically around the relationship between the outline and the gap between it and the inner circle. While reviewing we decided this should be adjusted as this issue is specifically about consistency with the design.

| Plugin | Design |
| --- | --- |
| <img width="239" alt="image" src="https://user-images.githubusercontent.com/1621608/191798727-7ed205ca-eaa7-4640-9a65-e5d16ddad45a.png"> | <img width="239" alt="image" src="https://user-images.githubusercontent.com/1621608/191799372-64fbadaa-8c0a-426a-9aad-14f97f05566f.png"> |
| <img width="315" alt="image" src="https://user-images.githubusercontent.com/1621608/191798284-714d9d93-b9aa-4234-9834-2538c732c293.png"> | <img width="671" alt="image" src="https://user-images.githubusercontent.com/1621608/191790471-2a31d44d-8e24-48dd-9789-6c15c20f7460.png"> |

**Comment by aaemnnosttv** [comment]
@nfmohit the difference in outline thickness has been fixed but there's still a difference in the gap around the inner content. Because this is an image radio, we can't change the shared style, but we could add an override for the color option specifically.

![image](https://user-images.githubusercontent.com/1621608/191847413-23bbff92-e8d8-4beb-8c15-7bf897e95a74.png)

We shouldn't sink too much time into making this perfect, but can we fix this quickly?

**Comment by aaemnnosttv** [comment]
My mistake – something was out of sync locally. All good now! Thanks @nfmohit 👍
